flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/SubtasksTimesInfo.java

```diff
@@ -132,13 +138,14 @@ public boolean equals(Object o) {
             SubtaskTimeInfo that = (SubtaskTimeInfo) o;
             return subtask == that.subtask
                     && Objects.equals(host, that.host)
+                    && Objects.equals(endpoint, that.endpoint)
```

Comment:
Can I confirm that the old and new objects are intended never to be equal - as the new objects have endpoint, does this have any implications?

Reply:
It's intended. Old objects and new objects are not intended to be compared anyway as they are essentially two different classes from two different versions of Flink release.